### PR TITLE
docs: update documentation for handling recipe changes

### PR DESCRIPTION
**Pull request by prince-chrismc**
Docs! 

~Follow up from the converation in https://github.com/conan-io/conan-center-index/pull/12468#issuecomment-1231981759~

This grew in scope. There going to be a new blog with the same information and we are trying to use the same language here.

I also found the "how to use rrev + locks" very generic so I am suggesting moving it to the conan docs and link from here https://github.com/conan-io/docs/pull/2741

### CODE REVIEW

**Review by uilianries — APPROVED** [review]

File: docs/consuming_recipes.md [hunk 1]
```diff
@@ -43,13 +43,17 @@ There can be several root causes if a recipe (a new revision) stopped to work in
...
+
+For larger projects and teams it is recommended to add some infrastructure to ensure stability by
 
  * **Cache recipes in your own Artifactory**: your project should use only this remote and
    new recipe revisions are only pushed to your Artifactory after they have been validated
    in your project.
- * **Pin the version of every reference you consume in your project** using recipe revisions
-   and lockfiles.
 
 ### Use your own Artifactory instance
```
Comment (prince-chrismc) [inline docs/consuming_recipes.md:58]: https://docs.conan.io/en/latest/uploading_packages/artifactory/artifactory_ce.html

**Comment by prince-chrismc** [comment]
Since I am working on the blog post that covers this I will circle back with the latest language

**Comment by prince-chrismc** [comment]
Once https://github.com/conan-io/docs/pull/2741 is merged, I will circle back with a link to that as well 😄

File: docs/consuming_recipes.md [hunk 2]
```diff
@@ -14,115 +14,57 @@ new recipe revisions stop to work in your project.
...
 
- * Create lockfile from `conanfile.txt` file:
+- [Running your own Conan Server](https://docs.conan.io/en/latest/uploading_packages/running_your_server.html) - great for local ad-hoc setups
+- [Cache recipes in your own ArtifactoryCE](https://docs.conan.io/en/latest/uploading_packages/using_artifactory.html) - recommended for production environments
 
-   ```sh
-   conan lock create conanfile.txt --lockfile-out=locks/project.lock
-   ```
+Using your own ArtifactoryCE instance is easy. You can [deploy it on-premise](https://conan.io/downloads.html) or use a 
+[cloud provided solution](https://jfrog.com/start-free/?isConan=true) for **free**. Your project should
```
Comment (memsharded) [inline docs/consuming_recipes.md:53]:
```suggestion
[cloud provided solution](https://jfrog.com/community/start-free) for **free**. Your project should 
```

**Review by memsharded — DISMISSED** [review]

**Review by SSE4 — APPROVED** [review]

**Pull request merged**